library/src/scala/quoted/Liftable.scala

```diff
@@ -43,4 +45,195 @@ object Liftable {
     }
   }
 
+  given [T: Type: Liftable: ClassTag] as Liftable[IArray[T]] = new Liftable[IArray[T]] {
+    def toExpr(iarray: IArray[T]): given QuoteContext => Expr[IArray[T]] = '{
+      val array = new Array[T](${iarray.length.toExpr})(ClassTag(${the[ClassTag[T]].runtimeClass.toExpr}))
+      ${ Expr.block(List.tabulate(iarray.length)(i => '{ array(${i.toExpr}) = ${iarray(i).toExpr} }), '{ array.asInstanceOf[IArray[T]] }) }
+    }
+  }
+
+  given [T: Type: Liftable] as Liftable[Seq[T]] = new Liftable[Seq[T]] {
+    def toExpr(seq: Seq[T]): given QuoteContext => Expr[Seq[T]] = given qctx => {
+      import qctx.tasty._
+      Repeated(seq.map(x => the[Liftable[T]].toExpr(x).unseal).toList, the[quoted.Type[T]].unseal).seal.asInstanceOf[Expr[Seq[T]]]
+    }
+  }
+
+  given [T: Type: Liftable] as Liftable[List[T]] = new Liftable[List[T]] {
+    def toExpr(x: List[T]): given QuoteContext => Expr[List[T]] = x match {
+      case x :: xs  => '{ (${xs.toExpr}).::[T](${x.toExpr}) }
+      case Nil => '{ Nil: List[T] }
+    }
+  }
+
+  given [T: Type: Liftable] as Liftable[Set[T]] = new Liftable[Set[T]] {
+    def toExpr(set: Set[T]): given QuoteContext => Expr[Set[T]] =
+      '{ Set(${set.toSeq.toExpr}: _*) }
+  }
+
+  given [T: Type: Liftable, U: Type: Liftable] as Liftable[Map[T, U]] = new Liftable[Map[T, U]] {
+    def toExpr(map: Map[T, U]): given QuoteContext => Expr[Map[T, U]] =
+    '{ Map(${map.toSeq.toExpr}: _*) }
+  }
+
+  given [T: Type: Liftable] as Liftable[Option[T]] = new Liftable[Option[T]] {
+    def toExpr(x: Option[T]): given QuoteContext => Expr[Option[T]] = x match {
+      case Some(x) => '{ Some[T](${x.toExpr}) }
+      case None => '{ None: Option[T] }
+    }
+  }
+
+  given [L: Type: Liftable, R: Type: Liftable] as Liftable[Either[L, R]] = new Liftable[Either[L, R]] {
+    def toExpr(x: Either[L, R]): given QuoteContext => Expr[Either[L, R]] = x match {
+      case Left(x) => '{ Left[L, R](${x.toExpr}) }
+      case Right(x) => '{ Right[L, R](${x.toExpr}) }
+    }
+  }
+
+  given [T1: Type: Liftable] as Liftable[Tuple1[T1]] = new {
+    def toExpr(tup: Tuple1[T1]) =
+      '{ Tuple1(${tup._1.toExpr}) }
+  }
```

Comment:
Minor: maybe still write `new Liftable[...]` to be consistent and explicit as above, the same for the code below

Reply:
It got way too long and made it harder to read.

Reply:
Specially for Tuple22